src/pkgcore/ebuild/processor.py

```diff
@@ -371,25 +372,37 @@ def __init__(self, userpriv, sandbox, fd_pipes=None):
             "PKGCORE_EBD_WRITE_FD": str(max_fd - 3),
         })
 
+        cread, cwrite, dread, dwrite = None
         # open pipes used for communication
-        cread, cwrite = os.pipe()
-        dread, dwrite = os.pipe()
-
-        # allow pipe overrides except ebd-related
-        ebd_pipes = {0: 0, 1: 1, 2: 2}
-        if fd_pipes:
-            ebd_pipes.update(fd_pipes)
-        ebd_pipes[max_fd - 4] = cread
-        ebd_pipes[max_fd - 3] = dwrite
-
-        # force each ebd instance to be a process group leader so everything
-        # can be easily terminated
-        self.pid = spawn_func(
-            [spawn.BASH_BINARY, self.ebd, "daemonize"],
-            fd_pipes=ebd_pipes, returnpid=True, env=env, pgid=0, **spawn_opts)[0]
-
-        os.close(cread)
-        os.close(dwrite)
+        try:
```

Comment:
a second set of eyes on the logic for cleanup/suppression would be appreciated- this is extremely edge case to trigger, I'm fixing it because I can see it in the code, not because I've managed to trigger it.

Reply:
Thinking about all the possible flows, it looks like you are super careful and catch all the things. I was checking for an exception in each statement, and I couldn't find a miss.